triedb/firewood/database.go

```diff
@@ -584,7 +584,7 @@ func arrangeKeyValuePairs(nodes *trienode.MergedNodeSet) ([][]byte, [][]byte) {
 	}
 
 	// We need to do all storage operations first, so prefix-deletion works for accounts.
-	keys := append(storageKeys, acctKeys...)
-	values := append(storageValues, acctValues...)
-	return keys, values
+	storageKeys = append(storageKeys, acctKeys...)
+	storageValues = append(storageValues, acctValues...)
+	return storageKeys, storageValues
```

Comment:
I don't really like having the variables referencing something incorrect.. I think we should either nolint this or directly return the results of the append.
```suggestion
	return append(storageKeys, acctKeys...), append(storageValues, acctValues...)
```

Technically we could use `slices.Concat` here... But that would introduce another copy of the slices that we probably don't want to do in prod code.